src/app/(proper_react)/(onboarding)/user/welcome/EnterInfo.tsx

```diff
@@ -4,22 +4,323 @@
 
 "use client";
 
+import Image from "next/image";
 import { Session } from "next-auth";
-import styles from "./EnterInfo.module.scss";
+import { FormEvent, useState } from "react";
+import { useOverlayTriggerState } from "react-stately";
+import { useOverlayTrigger } from "react-aria";
+import whyWeNeedInfoHero from "./images/welcome-why-we-need-info.svg";
 import { useL10n } from "../../../../hooks/l10n";
+import { ModalOverlay } from "../../../../components/client/dialog/ModalOverlay";
+import { Dialog } from "../../../../components/client/dialog/Dialog";
+import { Button } from "../../../../components/server/Button";
+import { InputField } from "../../../../components/client/InputField";
+
+import enterInfoStyles from "./EnterInfo.module.scss";
+import viewStyles from "./View.module.scss";
 
 export type Props = {
   onDataSaved: () => void;
+  onGoBack: () => void;
   user: Session["user"];
 };
 
-export const EnterInfo = (_props: Props) => {
-  const _l10n = useL10n();
+// TODO: Add more sophisticated validation for location data
```

Comment:
I think this doesn't need to be addressed here, but just FYI we'll need to parse the city/state separately (see the `ProfileData` object that's landed on `main` now for an idea of what the JSON form will need to look like).